packages/cli/src/Server.ts

```diff
@@ -195,33 +210,16 @@ export class Server extends AbstractServer {
 		setupPushHandler(restEndpoint, app);
 
 		if (config.getEnv('executions.mode') === 'queue') {
+			const { Queue } = await import('@/Queue');
 			await Container.get(Queue).init();
 		}
 
 		await handleMfaDisable();
 
-		await this.registerControllers();
-
-		// ----------------------------------------
-		// SAML
-		// ----------------------------------------
+		await this.registerAdditionalControllers();
 
-		// initialize SamlService if it is licensed, even if not enabled, to
-		// set up the initial environment
-		try {
-			await Container.get(SamlService).init();
-		} catch (error) {
-			this.logger.warn(`SAML initialization failed: ${error.message}`);
-		}
-
-		// ----------------------------------------
-		// Source Control
-		// ----------------------------------------
-		try {
-			await Container.get(SourceControlService).init();
-		} catch (error) {
-			this.logger.warn(`Source Control initialization failed: ${error.message}`);
-		}
+		// register all known controllers
+		Container.get(ControllerRegistry).activate(app);
```

Comment:
The order in which these are registered has been inversed here. Is that intentional or does it matter?

Reply:
here the controller used to be registered before the `init` on the related service had been called, which likely made some of those endpoint 5xx for a short while after startup. This change is meant to ensure that all relevant services have been initialized before we setup any of the routers.